tests/snuba/api/endpoints/test_organization_events_spans_performance.py

```diff
@@ -1033,6 +1033,69 @@ def test_bad_span_param(self):
             ]
         }
 
+    def test_span_filters(self):
+        spans = [
+            {
+                "same_process_as_parent": True,
+                "parent_span_id": "a" * 16,
+                "span_id": x * 16,
+                "start_timestamp": iso_format(self.min_ago + timedelta(seconds=1)),
+                "timestamp": iso_format(self.min_ago + timedelta(seconds=4)),
+                "op": "django.middleware",
+                "description": "middleware span",
+                "hash": "cd" * 8,
+                "exclusive_time": 3.0,
+            }
+            for x in ["b", "c"]
+        ]
+        event = self.create_event(spans=spans)
+
+        # same op different hash
+        different_hash_spans = [
+            {
+                "same_process_as_parent": True,
+                "parent_span_id": "a" * 16,
+                "span_id": x * 16,
+                "start_timestamp": iso_format(self.min_ago + timedelta(seconds=1)),
+                "timestamp": iso_format(self.min_ago + timedelta(seconds=4)),
+                "op": "django.middleware",
+                "description": "middleware span",
+                "hash": "ef" * 8,
+                "exclusive_time": 1.0,
+            }
+            for x in ["d", "e", "f"]
+        ]
+        self.create_event(spans=different_hash_spans)
+
+        different_op_spans = [
+            {
+                "same_process_as_parent": True,
+                "parent_span_id": "a" * 16,
+                "span_id": x * 16,
+                "start_timestamp": iso_format(self.min_ago + timedelta(seconds=1)),
+                "timestamp": iso_format(self.min_ago + timedelta(seconds=4)),
+                "op": "django.view",
+                "description": "view span",
+                "hash": "cd" * 8,
+                "exclusive_time": 1.0,
+            }
+            for x in ["d", "e", "f"]
+        ]
+        self.create_event(spans=different_op_spans)
+
+        with self.feature(self.FEATURES):
+            response = self.client.get(
+                self.url,
+                data={"project": self.project.id, "span": f"django.middleware:{'cd' * 8}"},
+                format="json",
+            )
+
+        assert response.status_code == 200, response.content
+
+        self.assert_span_examples(
+            response.data, [self.span_example_results("django.middleware", event)]
+        )
```

Comment:
I'm testing here that events with `spans=different_hash_spans` and `spans=different_op_spans` aren't returned in the response as example events and only the original event (that has the correct `op` and `hash`) is returned. Not sure if this is sufficient here 🤔

Reply:
I think this test somewhat misses the core of the fix. The fix here is to address spans that have the correct `op` and the correct `hash` but exist on 2 separate spans rather than a single span.

Neither `different_hash_spans` nor `different_op_spans` satisfy this condition so even if you remove the fix, I would expect this test to pass.

What you could try here is to create an event that contains 2 span
- `django.middleware:abababababababab`
- `django.view:cdcdcdcdcdcdcdcd`

Then try to filter for `django.middleware:cdcdcdcdcdcdcdcd` and the result should contain no events with the fix, and 1 event without the fix.

Reply:
that makes sense 🤦‍♀️ I was generating three different events lol